### ISSUE TITLE
Way to tell if kernel supports the features offered

### ISSUE THREAD

**Issue by Insomniac05** [Kingsman44/Pixelify#246]
Feature Description:
There is the note saying that magic editor and some other features are dependent on the device kernel, it would be great if there was a way to find out if the kernel supports it or not

**Comment by browntownington** [comment]
I'm seeking the same information. What kernel flags are required?
There's zero information on the web or telegram chat that I can see that says automatic call screening is reliant on the kernel version or flags. Only in the pixelify documentation does it state :

"New Automatic Call Screening depends on Device, Kernel."

If we know what flags are required we can seek a customer kernel, build our own, or acquire a device that has a supported kernel. Information is power 💪

So automatic call screening works only in the US and on Pixel 6 and 7 (at least from what I see [here](https://support.google.com/phoneapp/answer/9118387?hl=en)
My brother in australia has a Pixel 7 and he has done the mods for automatic call screening yet it doesn't work. So is google shipping a different kernel in the US and Australia? If so the pixel 7 source code is available [here ](https://android.googlesource.com/device/google/pantah-kernel/+refs)- we could compile the kernel with the right settings and we get this feature working.

**Comment by fillwithjoy1** [comment]
> I'm seeking the same information. What kernel flags are required? There's zero information on the web or telegram chat that I can see that says automatic call screening is reliant on the kernel version or flags. Only in the pixelify documentation does it state :
> 
> "New Automatic Call Screening depends on Device, Kernel."
> 
> If we know what flags are required we can seek a customer kernel, build our own, or acquire a device that has a supported kernel. Information is power 💪
> 
> So automatic call screening works only in the US and on Pixel 6 and 7 (at least from what I see [here](https://support.google.com/phoneapp/answer/9118387?hl=en) My brother in australia has a Pixel 7 and he has done the mods for automatic call screening yet it doesn't work. So is google shipping a different kernel in the US and Australia? If so the pixel 7 source code is available [here ](https://android.googlesource.com/device/google/pantah-kernel/+refs)- we could compile the kernel with the right settings and we get this feature working.

how is the progress going, even im annoyed that australians dont get auto call screning

**Comment by browntownington** [comment]
I'm struggling to be convinced it's kernel dependent until someone can provide evidence otherwise.

Reason being.. I looked at the pixel factory images and previous USA carriers and Australian carriers had different images. However all recent images appear to be the same regardless of carrier or country.

So from this I concluded they all run the same kernel. I'm going to try source a pixel and flash it with old US firmware and see if it works.

However I suspect it won't. 

Second reason I believe automatic call screening is not kernel dependent is; call screening works with pixelify on non pixel phones, as does call recording. The phone dialer detects spam calls on non pixel phones. The phone simply needs to call an indent or action to initiate call screening when the call is identified as spam etc. That's all it needs to do. I can't see how a particular kernel or parameter of the kernel would be required for this.

I believe automatic call screening is region dependent (USA). How it determines this I don't know yet.

Possibly it's to do with the region play store account is signed in with. Or possibility it looks at the carrier.

I'm clutching at straws speculating.

I posted in the telegram chat but now one responded.

**Comment by fillwithjoy1** [comment]
> I'm struggling to be convinced it's kernel dependent until someone can provide evidence otherwise.
> 
> Reason being.. I looked at the pixel factory images and previous USA carriers and Australian carriers had different images. However all recent images appear to be the same regardless of carrier or country.
> 
> So from this I concluded they all run the same kernel. I'm going to try source a pixel and flash it with old US firmware and see if it works.
> 
> However I suspect it won't.
> 
> Second reason I believe automatic call screening is not kernel dependent is; call screening works with pixelify on non pixel phones, as does call recording. The phone dialer detects spam calls on non pixel phones. The phone simply needs to call an indent or action to initiate call screening when the call is identified as spam etc. That's all it needs to do. I can't see how a particular kernel or parameter of the kernel would be required for this.
> 
> I believe automatic call screening is region dependent (USA). How it determines this I don't know yet.
> 
> Possibly it's to do with the region play store account is signed in with. Or possibility it looks at the carrier.
> 
> I'm clutching at straws speculating.
> 
> I posted in the telegram chat but now one responded.

yeah i have tried to get it working but its annoyingly hard. I think this could also be due to extra files that have to be implemented by google play store, but i have used gappsmod and even that doesn't solve the issue so idk
